webapp/src/store/Store.ts

```diff
@@ -1,7 +1,82 @@
+import fs from 'fs/promises';
+
 import { ProjectStore } from '@/store/ProjectStore';
+import MessageAdapterFactory from '@/utils/adapters/MessageAdapterFactory';
+import YamlTranslationAdapter from '@/utils/adapters/YamlTranslationAdapter';
+import { LyraProjectConfig } from '@/utils/lyraConfig';
+import { StoreData } from './types';
+
+const FILE_PATH = './store.json';
+
+/**
+ * Either we promised to complete initialization,
+ * or we are ready to initialize.
+ *
+ * Every concurrent task can await the same initialization.
+ */
+let promise: Promise<Store> | null = null;
```

Comment:
Nope, keeping state in module variables does not work in Next. It seems server actions use a separate module tree, perhaps only in dev mode.